generate-workflow.py

```diff
@@ -323,92 +324,173 @@ def __str__(self):
         return textwrap.indent(part1 + part2, 6 * ' ')
 
 
+class ActionStep:
+    def __init__(self, act_name, use_act, act_args):
+        self.act_name = act_name
+        self.use_act = use_act
+        self.act_args = act_args
+
+    def __str__(self):
+        part1 = textwrap.dedent(f'''\
+        - name: {self.act_name}
+          uses: {self.use_act}
+          with: 
+        ''')
+        all_args = ''
+        for arg_key in self.act_args:
+            all_args += arg_key + ": " + f'''{self.act_args[arg_key]}''' + '\n'
+        part2 = textwrap.indent(all_args, 4 * ' ')
+        return textwrap.indent(part1 + part2, 6 * ' ')
+
+
 def ensure_trailing_newline(s: str):
     return s + '\n' if s != '' and not s.endswith('\n') else s
 
 
-with open('.github/workflows/main.yml', 'w') as out:
-    out.write(HEADER)
-    for binfo in benchmarks:
-        for alltypes in (False, True):
-            at_dir = ('${{env.benchmark_conv_dir}}/' +
-                      ('alltypes' if alltypes else 'no-alltypes'))
-            at_job = 'alltypes' if alltypes else 'no_alltypes'
-            at_job_friendly = '-alltypes' if alltypes else 'no -alltypes'
-            convert_extra = (ensure_trailing_newline(binfo.convert_extra)
-                             if binfo.convert_extra is not None else '')
-            build_converted_cmd = binfo.build_converted_cmd.rstrip('\n')
-            # Python argparse thinks `--extra-3c-arg -alltypes` is two options
-            # rather than an option with an argument.
-            at_flag = '--extra-3c-arg=-alltypes \\\n' if alltypes else ''
-            at_ignore_step = ' (ignore failure)' if alltypes else ''
-            at_ignore_code = ' || true' if alltypes else ''
-
-            out.write(f'''\
-
-  test_{binfo.name}_{at_job}:
-    name: Test {binfo.friendly_name} ({at_job_friendly})
+def create_benchmark(out, binfo, alltypes, generate_stats=False, extra_args=[]):
+    # Python argparse thinks `--extra-3c-arg -alltypes` is two options
+    # rather than an option with an argument.
+    at_flag = '--extra-3c-arg=-alltypes \\\n' if alltypes else ''
+
+    job_suffix = ''
+    for earg in extra_args:
+        if not job_suffix:
+            job_suffix = "_"
+        at_flag += '--extra-3c-arg=' + earg + ' \\\n'
+        job_suffix += earg.replace('-', '_')
+
+    at_dir = ('${{env.benchmark_conv_dir}}/' +
```

Comment:
`at` here was short for `alltypes`.  If this variable is going to be generalized to include information not related to `alltypes`, its name should be updated.